server/src/main/java/org/apache/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -267,6 +275,12 @@ public AppenderatorAddResult add(
       throw new SegmentNotWritableException("Attempt to add row to swapped-out sink for segment[%s].", identifier);
     }
 
+    if (addResult.isRowAdded()) {
+      rowIngestionMeters.incrementProcessed();
+    } else if (addResult.hasParseException()) {
+      parseExceptionHandler.handle(addResult.getParseException());
+    }
```

Comment:
Test coverage bot is complaining about a lack of coverage for this condition. I think we should add tests for these branches. Right now `AppenderatorTest` only tests `addResult.isRowAdded`

Covering all 4 branches should make the code coverage bot happy too.